[Bug]: tests/integration/v2/go.mod requires /v2 module at invalid version v0.0.0, breaking mocked integration CI

### SDK Version

main @ 62cccae9a1d1c665291eb35f49bb4a52cb52a7d8 (post-#644 merge); latest published release: v2.0.0

### Go Version

go1.27.0 (CI `stable` leg; failure is Go-version-independent — the go.mod parse error occurs under any toolchain)

### Module

other / not sure

### What happened?

The **Integration Tests (mocked)** CI job fails immediately with a `go.mod` parse error, before any integration test runs:

```
go: errors parsing go.mod:
go.mod:9:2: require github.com/michaeldcanady/servicenow-sdk-go/v2: version "v0.0.0" invalid: should be v2, not v0
```

`tests/integration/v2/go.mod` line 9 requires the SDK as:

```
github.com/michaeldcanady/servicenow-sdk-go/v2 v0.0.0
```

For a module path with a `/v2` suffix, the go tool rejects any required version below `v2`. The `replace github.com/.../v2 => ../../../` directive on line 40 redirects resolution to the local checkout, but it does **not** exempt the `require` line from version validation — parsing fails first.

This is not caused by PR #644's code: the identical failure appears on main's CI run 32211674551 (Aug 19, commit acc85be / v2.0.0), on PR #644's run 32425391807, and on the post-merge push run.

### What did you expect?

`cd tests/integration/v2 && go test -tags integration ./...` parses the module and runs the godog suite, and the **Integration Tests (mocked)** check goes green.

### Reproduction

```go
// No SDK code involved — reproduce from a clean checkout of main:
//   git clone https://github.com/michaeldcanady/servicenow-sdk-go && cd servicenow-sdk-go
//   cd tests/integration/v2
//   go test -tags integration ./...
//
// Or in CI, the same command the job runs:
//   cd tests/integration/v2
//   go test -tags integration ./...   # with SN_OFFLINE=true + mock creds, per ci.yml
```

### Error output (if any)

```shell
go: errors parsing go.mod:
go.mod:9:2: require github.com/michaeldcanady/servicenow-sdk-go/v2: version "v0.0.0" invalid: should be v2, not v0
##[error]Process completed with exit code 1.
```

Job log: https://github.com/michaeldcanady/servicenow-sdk-go/actions/runs/32425391807/job/96630247665

### Additional context

- Suggested fix: bump the require to a valid v2 version (e.g. `github.com/michaeldcanady/servicenow-sdk-go/v2 v2.0.0`). The existing `replace` directive still redirects to the local checkout, so no other change should be needed.
- Related: #624 (add integration test job to CI), #560 (CI coverage gaps epic).
- First observed failing on main's Aug 19 run (32211674551); still failing after #644 merged (62cccae).

--- a/tests/integration/v2/go.mod
+++ b/tests/integration/v2/go.mod
@@ -1,4 +1,4 @@
-module github.com/michaeldcanady/servicenow-sdk-go/tests/integration/v2
+module github.com/michaeldcanady/servicenow-sdk-go/v2/tests/integration/v2
 
 go 1.25.0
 
@@ -6,7 +6,7 @@ require (
 	github.com/cucumber/godog v0.15.1
 	github.com/jarcoal/httpmock v1.4.1
 	github.com/joho/godotenv v1.5.1
-	github.com/michaeldcanady/servicenow-sdk-go/v2 v0.0.0
+	github.com/michaeldcanady/servicenow-sdk-go/v2 v2.0.0-00010101000000-000000000000
 	github.com/microsoft/kiota-abstractions-go v1.9.4
 	github.com/microsoft/kiota-http-go v1.5.6
 )

